ramls/oai-pmh.raml

```diff
@@ -0,0 +1,131 @@
+#%RAML 1.0
+title: OAI-PMH API
+baseUri: https://github.com/folio-org/mod-oai-pmh
+version: v1
+protocols: [ HTTP, HTTPS ]
+
+documentation:
+  - title: OAI-PMH Business Logic API
+    content: |
+      The Open Archives Initiative Protocol for Metadata Harvesting (OAI-PMH) provides
+      an application-independent interoperability framework based on metadata harvesting.
+      This module supports the OAI-PMH as a means of exposing FOLIO metadata.
+
+types:
+  oai-pmh: !include schemas/OAI-PMH.xsd
+
+traits:
+  partionable: !include traits/partitionable.raml
+  selective: !include traits/selective.raml
+  metadataPrefix: !include traits/metadataPrefix.raml
+
+resourceTypes:
+  list: !include rtypes/list.raml
+
+/oai:
+  /records:
+    displayName: Records
+    type:
+      list:
+        exampleCollection: !include examples/records.sample
+    get:
+      is: [ partionable, selective, metadataPrefix ]
+      responses:
+        404:
+          description: "Item with a given ID and selective harvesting params (if any) not found"
+          body:
+            application/xml:
+              schema: oai-pmh
+              example: !include examples/no_records_error.sample
+
+    /{id}:
+      uriParameters:
+        id:
+          description: The unique identifier of the item in the repository
+          type: string
+      get:
```

Comment:
I think you might need a 422 here too in case some unknown metadataprefix is specified (cannotDisseminateFormat Error)

Reply:
/records/{id} resource has 'metadataPrefix' trait which defines 'metadataPrefix' query param _as well as 422 status code_